utils/small_vector.hh

```diff
@@ -442,12 +443,15 @@ public:
         std::swap(*this, other);
     }
 
-    bool operator==(const small_vector& other) const noexcept {
-        return size() == other.size() && std::equal(_begin, _end, other.begin());
```

Comment:
I think we should keep operator==, as it's faster (depending on whether T::operator== is faster than T::operator<=> or not).

Reply:
Also, we'll lose operator== if T isn't three-way comparable but is equality comparable.

Reply:
Oh, I see you did keep operator==, it's just below.